src/main/java/seedu/address/model/DebtsTable.java

```diff
@@ -52,8 +53,27 @@ public void add(Person personToAdd) {
         this.putIfAbsent(personToAdd, new DebtsList());
     }
 
-    public DebtsTable asObservableList() {
-        return internalList;
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedPerson}.
+     *
+     */
+    public void setPerson(Person target, Person editedPerson) throws PersonNotFoundException, DuplicatePersonException {
+        requireNonNull(editedPerson);
+        requireNonNull(target);
+        if (!this.containsKey(target)) {
+            throw new PersonNotFoundException();
+        }
+        if (!target.equals(editedPerson) && this.containsKey(editedPerson)) {
+            throw new DuplicatePersonException();
```

Comment:
I think you should create a new exception called PersonNameNotChangedException() since DuplicatePersonException() sends a message that doesn't make sense here.

Reply:
No, here this means that editedPerson is already in the addressbook, and it is not the targeted person

Reply:
So if we change target to editedPerson, a DuplicatePersonException will occurs.

Reply:
It does ! This means that editedPerson is already in the addressbook, so if you change target person into editedPerson, the "Operation would result in duplicate persons", which is the message of DuplicatePersonException().

Reply:
Oh ok thank you for clarifying